app/api/custom/orders.py

```diff
@@ -184,3 +189,33 @@ def ticket_attendee_pdf(attendee_id):
     if not os.path.isfile(file_path):
         create_pdf_tickets_for_holder(ticket_holder.order)
     return send_from_directory('../', file_path, as_attachment=True)
+
+@order_blueprint.route('/<string:order_identifier>/verify', methods=['POST'])
+def verify_order_payment(order_identifier):
+
+    order = Order.query.filter_by(identifier=order_identifier).first()
+    
+    try:
+        session = StripePaymentsManager.retrieve_session(order.event, order.stripe_session_id)
+        payment_intent = StripePaymentsManager.retrieve_payment_intent(order.event, session.payment_intent)
+    except Exception as e:
+        raise e
+
+    if session['payment_status'] == 'paid':
+        order.status = 'completed'
+        order.completed_at = datetime.utcnow()
+        order.paid_via = payment_intent['charges']['data'][0]['payment_method_details']['type']
+        order.transaction_id = payment_intent['charges']['data'][0]['balance_transaction']
+        save_to_db(order)
+
+        on_order_completed(order)
+    
+    if session['payment_status'] == 'unpaid' and payment_intent['status'] != 'succeeded':
```

Comment:
This should be else case as we hit only when the user has returned from Stripe.

Reply:
This part is to be removed as suggested by Mario in last meet.

Reply:
that means? why?

Reply:
Mario said we should do nothing when user clicks back button and should keep the order active until it expires automatically.